### PR TITLE
[terra-date-picker] Fix Responsive Resize issue

### PR DESCRIPTION
**Pull request by ashishkcerner**
### Summary

**What was changed:**
DatePicker responsive screen fix upon resize

**Why it was changed:**
While zooming through the screen datepicker was not accessible properly

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9480 

**Pre implementation.**
Screenshot 2023-08-31 at 11 09 27 AM

**Post implementation.**

https://github.com/cerner/terra-framework/assets/126875089/5c79b941-88af-4451-967e-fd1f497f8cad

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by supreethmr** [comment]
@ashishkcerner  I am not able to see the today button of calendar on PostImplementation video is it still inaccessible with 400% zoom.

File: packages/terra-date-picker/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Changelog
 
 ## Unreleased
-
+* Fixed
```
Comment (supreethmr) [inline packages/terra-date-picker/CHANGELOG.md:4]: Add a line break for consistency
```suggestion

* Fixed
```

File: packages/terra-date-picker/src/react-datepicker/stylesheets/react_datepicker.module.scss [hunk 2]
```diff
@@ -430,7 +430,13 @@
       border: 1.8 * $datepicker-navigation-size solid transparent;
     }
   }
-
+  @media (max-width: 320px), (max-height: 256px) {
+    .react-datepicker-portal {
+      align-items: flex-start;
+      overflow: auto;
+    }
+  }
```
Comment (supreethmr) [inline packages/terra-date-picker/src/react-datepicker/stylesheets/react_datepicker.module.scss:438]: add a line break and comment for consistency 

```suggestion

  // Allows calendar to scroll on smaller viewports
  @media (max-width: 320px), (max-height: 256px) {
    .react-datepicker-portal {
      align-items: flex-start;
      overflow: auto;
    }
  }
```

**Comment by ashishkcerner** [comment]
> @ashishkcerner I am not able to see the today button of calendar on PostImplementation video is it still inaccessible with 400% zoom.

Yes @supreethmr  have added the new video now .It is. accessible

**Review by supreethmr — APPROVED** [review]

File: packages/terra-date-picker/CHANGELOG.md (hunk 1 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/CHANGELOG.md:5]: Done https://github.com/cerner/terra-framework/commit/acd9d0dfd0f8928b39ceb4dedbb003c77806849d

File: packages/terra-date-picker/src/react-datepicker/stylesheets/react_datepicker.module.scss (hunk 2 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/react-datepicker/stylesheets/react_datepicker.module.scss:440]: Done https://github.com/cerner/terra-framework/commit/acd9d0dfd0f8928b39ceb4dedbb003c77806849d

**Review by sugan2416 — APPROVED** [review]

**Pull request merged**